source/extensions/filters/network/rocketmq_proxy/codec.cc

```diff
@@ -0,0 +1,293 @@
+#include "extensions/filters/network/rocketmq_proxy/codec.h"
+
+#include <string>
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "extensions/filters/network/rocketmq_proxy/protocol.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RocketmqProxy {
+
+const uint32_t RemotingCommand::SHIFT_RPC = 0;
+const uint32_t RemotingCommand::SHIFT_ONEWAY = 1;
+const uint32_t Decoder::MIN_FRAME_SIZE = 8;
+const uint32_t Decoder::FRAME_LENGTH_FIELD_SIZE = 4;
+const uint32_t Decoder::FRAME_HEADER_LENGTH_FIELD_SIZE = 4;
+
+RemotingCommandPtr Decoder::decode(Buffer::Instance& buffer, bool& underflow, bool& has_error,
+                                   int request_code) {
+  // Verify there is at least some bits, which stores frame length and header length
+  if (buffer.length() <= MIN_FRAME_SIZE) {
+    underflow = true;
+    return nullptr;
+  }
+
+  auto frame_length = buffer.peekBEInt<uint32_t>();
```

Comment:
is there a maximum value limit for this?

Reply:
No limitation here.